### PR TITLE
[Doc] Refactor Theming documentation

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problems

- Developers have to read the MUI doc about theming to learn react-admin styling
- The Theming documentation repeats tips found in other chapters, except that it's outdated and the other chapters are up to date

## Solution

A dedicated Theming doc section with several chapters. 

image

## To do

- [x] Split the Theming chapter into smaller chapters
- [x] Add screenshots
- [x] Update old links to the Theming chapter

### CODE REVIEW

File: packages/ra-ui-materialui/src/field/BooleanField.tsx [hunk 1]
```diff
@@ -49,12 +49,20 @@ const BooleanFieldImpl = <
                 <Tooltip title={translate(ariaLabel, { _: ariaLabel })}>
                     {isTruthyValue ? (
                         TrueIcon ? (
-                            <TrueIcon data-testid="true" fontSize="small" />
+                            <TrueIcon
+                                data-testid="true"
+                                fontSize="small"
+                                className={classes.trueIcon}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/BooleanField.tsx:55]: This listtle change was necessary for an example in the SX documentation

File: docs/css/style-v18.css
Comment (fzaninotto) [inline docs/css/style-v18.css:1]: These changes fix the TOC, which was too small once readers scroll down on a page and hide the app bar.

File: docs/Theming.md [hunk 2]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
 
 # Theming
 
-Whether you need to adjust a CSS rule for a single component, or change the color of the labels in the entire app, you're covered!
+React-admin applications use a neutral style by default. You will probably want to customize the look and feel to match your branding, or your end users preferences. Don't worry! You can customize the look and feel of pretty much everything in react-admin.
 
-## `sx`: Overriding A Component Style
```
Comment (fzaninotto) [inline docs/Theming.md:10]: All removed sections are present elsewhere. I checked ;)

File: docs/Theming.md [hunk 3]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
+        <Resource name="songs" list={SongList} />
+        <Resource name="artists" list={ArtistList} />
     </Admin>
 );
 ```
 
-**Tip:** [React's Error Boundaries](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) are used internally to display the Error Page whenever an error occurs. Error Boundaries only catch errors during rendering, in lifecycle methods, and in constructors of the components tree. This implies in particular that errors during event callbacks (such as 'onClick') are not concerned. Also note that the Error Boundary component is only set around the main container of React Admin. In particular, you won't see it for errors thrown by the [sidebar Menu](./Menu.md), nor the [AppBar](#customizing-the-appbar-content). This ensures the user is always able to navigate away from the Error Page.
+You can even write your own layout component from scratch. Check the [dedicated Layout chapter](./Layout.md) for more information.
 
-## Loading
```
Comment (fzaninotto) [inline docs/Theming.md:1086]: Except Loading and LinearProgress, which were poorly documente anyway

File: docs/AppBar.md [hunk 4]
```diff
@@ -273,7 +273,30 @@ const MyAppBar = () => (
...
+    />
+);
+
+const MyUserMenu = props => (<UserMenu {...props} icon={<MyCustomIcon />} />);
+
+const MyAppBar = () => <AppBar userMenu={<MyUserMenu />} />;
+```
+{% endraw %}
+
+Finally, can hide the user menu by setting the `userMenu` prop to `false`.
```
Comment (adguernier) [inline docs/AppBar.md:299]:
```suggestion
Finally, you can hide the user menu by setting the `userMenu` prop to `false`.
```

File: docs/Authentication.md [hunk 5]
```diff
@@ -295,11 +294,24 @@ const App = () => (
 );
 ```
 
-Use the `useLogin` hook in your custom `LoginPage` component.
+By default, the login page displays a gradient background. If you just want to change the background, you can use the default Login page component and pass an image URL as the `backgroundImage` prop.
+
+```jsx
+// in src/MyLoginPage.js
+import {  Login } from 'react-admin';
```
Comment (adguernier) [inline docs/Authentication.md:301]:
```suggestion
import { Login } from 'react-admin';
```

File: docs/Authentication.md [hunk 6]
```diff
@@ -295,11 +294,24 @@ const App = () => (
...
+
+const MyLoginPage = () => (
+    <Login
+        // A random image that changes everyday
+        backgroundImage="https://source.unsplash.com/random/1600x900/daily"
+    />
+);
+```
+
+If you want to build a Login page from scratch, you'll need the `useLogin` hook.
```
Comment (adguernier) [inline docs/Authentication.md:311]:
```suggestion
If you want to build a Login page from scratch, you'll need the [`useLogin` hook](./useLogin.md).
```

File: docs/Admin.md [hunk 7]
```diff
@@ -789,7 +787,7 @@ const App = () => (
 
 If you want to support both a light and a dark theme, check out [the `<Admin darkTheme>` prop](#darktheme). 
 
-For more details on predefined themes and custom themes, refer to [the Theming chapter](./Theming.md#global-theme-overrides) of the react-admin documentation.
+For more details on predefined themes and custom themes, refer to [the Application Theme chapter](./AppTheme.md).
```
Comment (djhi) [inline docs/Admin.md:790]:
```suggestion
For more details on predefined and custom themes, refer to [the Application Theme chapter](./AppTheme.md).
```

File: docs/Authentication.md (hunk 6 above)
Comment (djhi) [inline docs/Authentication.md:311]:
```suggestion
If you want to build a Login page from scratch, you'll need the [`useLogin`](./useLogin.md) hook.
```

File: docs/Layout.md [hunk 8]
```diff
@@ -370,8 +413,59 @@ export const MyLayout = (props) => (
...
+import { Box } from '@mui/material';
+import { AppBar, Menu, Sidebar } from 'react-admin';
+
+const MyLayout = ({ children, dashboard }) => (
+    <Box 
+        display="flex"
+        flexDirection="column"
+        zIndex={1}
+        minHeight="100vh"
+        backgroundColor='theme.palette.background.default,'
```
Comment (adguernier) [inline docs/Layout.md:441]:
```
backgroundColor='theme.palette.background.default,'
```
Is the ending comma necessary here ?

File: docs/List.md [hunk 9]
```diff
@@ -145,12 +148,16 @@ const ListActions = () => {
 
 ## `aside`
 
-You may want to display additional information on the side of the list. Use the `aside` prop for that, passing the component of your choice:
+The default `<List>` layout lets you render the component of your choice on the side of the list.
+
+![List with aside](./img/list_aside.webp)
+
+PAss a React element as the  `aside` prop for that purpose:
```
Comment (adguernier) [inline docs/List.md:155]:
```suggestion
Pass a React element as the `aside` prop for that purpose:
```

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/Theming.md [hunk 10]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
-    EditButton,
-    List,
-    NumberField,
-    TextField,
-    ShowButton,
-} from 'react-admin';
-import Icon from '@mui/icons-material/Person';
-
-export const VisitorIcon = Icon;
+import { Datagrid, TextField } from 'react-admin';
```
Comment (adguernier) [inline docs/Theming.md:27]:
```suggestion
import { Datagrid, TextField, List, DateField, ReferenceManyCount, BooleanField } from 'react-admin';
```

File: docs/Theming.md [hunk 11]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
-    const originalPathname = useRef(pathname);
+## Customizing The Page Layout
 
-    // Effect that resets the error state whenever the location changes
-    useEffect(() => {
-        if (pathname !== originalPathname.current) {
-            resetErrorBoundary();
-        }
-    }, [pathname, resetErrorBoundary]);
+React-admin provides alternative layout components. For instance, you can replace the default `<Layout>`, which uses a sidebar for navigation, with a `<ContainerLayout>`, which uses a top bar instead.
```
Comment (adguernier) [inline docs/Theming.md:125]:
```suggestion
React-admin provides alternative layout components. For instance, you can replace the default `<Layout>`, which uses a sidebar for navigation, with a [`<ContainerLayout>`](./ContainerLayout.md), which uses a top bar instead.
```

File: docs/Theming.md [hunk 12]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
 
-Usage:
+Material-ui is a rich UI library, so you may not know all its components. There are 3 components you should know about when theming react-admin:
 
-```jsx
-<Loading loadingPrimary="app.page.loading" loadingSecondary="app.message.loading" />
-```
-
-## LinearProgress
+- `<Box>`: a generic container component, which can be used to add padding, margin, and background color to any component. It is used extensively in react-admin, and you will probably use it a lot too.
```
Comment (adguernier) [inline docs/Theming.md:162]:
```suggestion
- [`<Box>`](https://mui.com/material-ui/react-box): a generic container component, which can be used to add padding, margin, and background color to any component. It is used extensively in react-admin, and you will probably use it a lot too.
```

File: docs/Theming.md [hunk 13]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
+- `<Stack>`: a component to stack elements vertically or horizontally.
```
Comment (adguernier) [inline docs/Theming.md:163]:
```suggestion
- [`<Stack>`](https://mui.com/material-ui/react-stack): a component to stack elements vertically or horizontally. 
```

File: docs/Theming.md [hunk 14]
```diff
@@ -5,1109 +5,164 @@ title: "Theming"
...
+Material-ui is a rich UI library, so you may not know all its components. There are 3 components you should know about when theming react-admin:
 
-```jsx
-<Loading loadingPrimary="app.page.loading" loadingSecondary="app.message.loading" />
-```
-
-## LinearProgress
+- `<Box>`: a generic container component, which can be used to add padding, margin, and background color to any component. It is used extensively in react-admin, and you will probably use it a lot too.
+- `<Stack>`: a component to stack elements vertically or horizontally. 
+- `<Grid>`: a component to display elements in a grid.
```
Comment (adguernier) [inline docs/Theming.md:164]:
```suggestion
- [`<Grid>`](https://mui.com/material-ui/react-grid): a component to display elements in a grid. 
```

File: docs/SX.md [hunk 15]
```diff
@@ -0,0 +1,467 @@
+---
+layout: default
+title: "SX"
+---
+
+# The `sx` prop
+
+All react-admin components expose an `sx` prop, which allows to customize the component style. This `sx` prop is kind of like [React's `style` prop](https://react.dev/reference/react-dom/components/common#applying-css-styles), but it's more powerful. It uses the CSS-in-JS solution offered by MUI, [MUI System](https://mui.com/system/basics/#the-sx-prop), powered by [emotion](https://emotion.sh/docs/introduction).
```
Comment (adguernier) [inline docs/SX.md:8]:
```suggestion
All react-admin components expose a `sx` prop, which allows to customize the component style. This `sx` prop is kind of like [React's `style` prop](https://react.dev/reference/react-dom/components/common#applying-css-styles), but it's more powerful. It uses the CSS-in-JS solution offered by MUI, [MUI System](https://mui.com/system/basics/#the-sx-prop), powered by [emotion](https://emotion.sh/docs/introduction). 
```

File: docs/SX.md [hunk 16]
```diff
@@ -0,0 +1,467 @@
...
+    // some styles
+    ":hover": {
+      boxShadow: 6,
+    },
+  }}
+>
+```
+{% endraw %}
+
+Common pseudo-classes used for styling are `:hover`, `:disabled`, `:active`, `:focus`, `:first-child`, `:last-child`. Check [the list of CDD pseudo classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) for more information.
```
Comment (adguernier) [inline docs/SX.md:241]:
```suggestion
Common pseudo-classes used for styling are `:hover`, `:disabled`, `:active`, `:focus`, `:first-child`, `:last-child`. Check [the list of CSS pseudo classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) for more information.
```

File: docs/AppTheme.md [hunk 17]
```diff
@@ -0,0 +1,450 @@
...
+- Don't forget to merge your custom style overrides with the ones from react-admin's `defaultTheme`, otherwise the application will have the default Material UI theme (most notably, outlined text inputs)
+- Custom style overrides must live under a `root` key. Then, the style override syntax is the same as the one used for the `sx` prop.
+
+To guess the name of the subclass to use (like `.RaDatagrid-headerCell` above) for customizing a component, you can use the developer tools of your browser, or check the react-admin documentation for individual components (e.g. the [Datagrid CSS documentation](./Datagrid.md#sx-css-api)).
+
+**Tip**: As an alternative, you can also re-export styled components, and use them instead of the react-admin components. Check the [Reusable Components](./SX.md#reusable-components) documentation for an example.
+
+## Overriding Default Props
+
+You can use this technique to override not only styles, but also defaults for components. That's how react-admin applies the `filled` variant to all `TextField` components. So for instance, to change the variant to `outlined`, create a custom theme as follows:
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/AppTheme.md:90]: no, my version is correct

File: docs/SX.md (hunk 15 above)
Comment (fzaninotto) [reply docs/SX.md:8]: No, my version is correct

https://www.quora.com/When-a-word-starts-with-an-%E2%80%9Cs%E2%80%9D-do-you-use-%E2%80%9Ca%E2%80%9D-or-%E2%80%9Can%E2%80%9D-before-the-word

**Review by djhi — APPROVED** [review]
Very nice !

**Pull request merged**